src/platforms/apple/common/configuration/watchdog-terminations.mdx

```diff
@@ -0,0 +1,48 @@
+---
+title: Watchdog Terminations
+sidebar_order: 10
+description: "Learn how to turn off Watchdog Termination tracking"
+---
+
+<Note>
+This integration is available since 8.0.0. It was named <PlatformLink to="/configuration/out-of-memory/">Out of Memory</PlatformLink> before the 8.0.0 release.
+</Note>
+
+This integration tracks <a href="https://developer.apple.com/documentation/xcode/addressing-watchdog-terminations">watchdog terminations</a> based on heuristics. This feature is available for iOS, tvOS, and Mac Catalyst, works only if the application was in the foreground, and doesn't track watchdog terminations for unit tests.
+
+When a typical unhandled error occurs, the Apple SDK writes a report to disk with the current state of your application, with details like the stack trace, tags, breadcrumbs, and so on, before the app terminates. When the watchdog terminates your app this happens without further notice, which means the SDK can't write a report to disk. A common reason the watchdog can terminate your app is an Out Of Memory problem. In the case the app is terminated because it hangs we don't create a watchdog termination event, instead an AppHangs event is created.
+
+As a result, in the Apple SDK, we track watchdog terminations during the app start based on heuristics, but getting the state of the app when a watchdog termination occurs is challenging. To provide the same state as we do for a typical crash, we would periodically need to store the app state to disk. This would cause a lot of I/O, which in turn could negatively affect the application's performance. To avoid this, watchdog termination events have less context than typical unhandled errors. Breadcrumbs are stored to disk and are added to watchdog termination events to help you figure out where in the app the event happened. This is done by appending the breadcrumbs to an open file and should have a marginal impact on your app's performance.
+
+When a user launches the app, the SDK checks the following statements and reports a watchdog termination if all of them are true:
+
+- The app didn't crash on the previous run.
+- The app was in the foreground/active.
+- The user launched the app at least once. When your app watch killed by the watchdog during the first launch, the SDK can't detect it.
+- There was no debugger attached to the process of the app. If there was, our logic would falsely report OOMs whenever you restarted the app in development.
```

Comment:
This bit is a little confusing. In the second sentence, are you describing the behaviour that will occur or the behaviour we're trying to avoid? If the latter, then I would put this in parentheses to make that clearer.
```suggestion
- There was no debugger attached to the process of the app. If there was, our logic would falsely report OOMs whenever you restarted the app in development.
```

Reply:
This is a direct copy from the existing OOM docs (https://docs.sentry.io/platforms/apple/configuration/out-of-memory/). To me it's obvious, but happy to see a suggestion, the suggested change here doesn't seem to actually change anything?

Reply:
I hadn't made the change since I wasn't sure what the meaning was. I'm still not clear on what it means, lol. Is this the expected behaviour or the behaviour we're trying to avoid?

Reply:
If the debugger was attached, we won't report watchdog terminations. Because if the debugger was attached we'd otherwise falsely report watchdog terminations. I've added the word "otherwise" in there which should clear things up I think.